### PR TITLE
PDFViewer: Multipage view, misc QOL changes

### PR DESCRIPTION
**Pull request by mattco98**
We now support displaying multiple pages in a single columns, and smoothly scrolling through them, much like any other PDF viewer. The user can toggle between single- and multi-page mode. 

Also fixes a couple bugs, such as re-rendering the pages on app resize and auto-sizing the page label (so that large total pages don't turn into `...`).

### CODE REVIEW

**Comment by linusg** [comment]
This has a conflict

**Comment by mattco98** [comment]
> This has a conflict

Fixed!

File: Userland/Applications/PDFViewer/PDFViewer.h [hunk 1]
```diff
@@ -18,7 +18,7 @@ class PDFViewer : public GUI::AbstractScrollableWidget {
     C_OBJECT(PDFViewer)
 
 public:
-    virtual ~PDFViewer() override = default;
+    ~PDFViewer() override = default;
```
Comment (awesomekling) [inline Userland/Applications/PDFViewer/PDFViewer.h:46]: This looks unrelated to the commit.

Also please don't change this. SerenityOS style is to specify both `virtual` *and* `override`/`final` for readability. :^)

**Comment by awesomekling** [comment]
Typo in commit message: propogate -> propagate

File: Userland/Applications/PDFViewer/PDFViewer.cpp [hunk 2]
```diff
@@ -247,3 +306,61 @@ PDF::PDFErrorOr<NonnullRefPtr<Gfx::Bitmap>> PDFViewer::render_page(const PDF::Pa
...
 }
+
+PDF::PDFErrorOr<void> PDFViewer::cache_page_dimensions(bool recalculate_fixed_info)
+{
+    if (recalculate_fixed_info)
+        m_page_dimension_cache.page_info.clear_with_capacity();
+
+    if (m_page_dimension_cache.page_info.is_empty()) {
+        m_page_dimension_cache.page_info.ensure_capacity(m_document->get_page_count());
+        for (size_t i = 0; i < m_document->get_page_count(); i++) {
```
Comment (nico) [inline Userland/Applications/PDFViewer/PDFViewer.cpp:324]: This forces us to read all page objects, of which there could be hundreds of thousands. This was identified as one of the problems in #23036. Could we instead compute information for pages lazily as needed, so we only have to read a few pages around the current page?

Comment (mattco98) [reply Userland/Applications/PDFViewer/PDFViewer.cpp:324]: Yeah I had started working on that, I can resurrect it if it looks like we do actually want to do that after all

Comment (nico) [reply Userland/Applications/PDFViewer/PDFViewer.cpp:324]: (To be clear, I meant do it lazily at the Applications/PDFViewer level. If that's what you mean too, then I think that'd be great to do!)

Comment (mattco98) [reply Userland/Applications/PDFViewer/PDFViewer.cpp:324]: Ahh, didn't really look at the code block you quoted closely enough. No that's not what I meant, I haven't done any work towards that
